python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -333,6 +358,31 @@ def _get_num_gpus(
     return None
 
 
+def _get_num_tpus(
+    ray_start_params: Dict[str, str],
+    k8s_resource_limits: Dict[str, Any],
+    group_name: str,
+) -> Optional[int]:
+    """Get memory resource annotation from ray_start_params or k8s_resource_limits,
+    with priority for ray_start_params.
+    """
+
+    if "num-tpus" in ray_start_params:
+        return int(ray_start_params["num-tpus"])
+    else:
+        for key in k8s_resource_limits:
+            # e.g. google.com/tpu
```

Comment:
I'm unfamiliar with the `k8s_resource_limits` - does this refer to how GKE selects TPUs or is it the resource string provided to ray?

I'm assuming it's the former, but if it's the latter could we standardize on the resource string being "TPU" rather than google.com/tpu? That's what [TPU resource autodetection uses](https://github.com/ray-project/ray/blob/master/python/ray/_private/accelerators/tpu.py#L78-L80)

Reply:
I believe it refers to the resource string provided to Ray, this is an example [here](https://github.com/ray-project/kuberay/blob/49a5725b2a5782aad8da8a715a0114523c792372/ray-operator/config/samples/ray-cluster.tpu-v4-multihost.yaml#L91). I think standardizing the naming makes sense, I'll add that change in.

Reply:
A lot of example yamls I see use `google.com/tpu` though, maybe it'd make sense to support both?

Reply:
Ah okay, I was thinking of something different, i.e. something like this:

```
 workerGroupSpecs:
  - rayStartParams:
      block: "true"
      resources: '"{\"TPU\": 4\"}"'
```

I think this `k8s_resource_limits` from the example you provided refers to a GKE specific key so there's a no op here